src/Sylius/Bundle/LocaleBundle/Form/DataTransformer/ArrayGregorianToCalendarSystemTransformer.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\LocaleBundle\Form\DataTransformer;
+
+use Sylius\Bundle\LocaleBundle\Templating\Helper\LocaleHelper;
+use Symfony\Component\Form\DataTransformerInterface;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+
+/**
+ * @author Aram Alipoor <aram.alipoor@gmail.com>
+ */
+class ArrayGregorianToCalendarSystemTransformer implements DataTransformerInterface {
+
+    /**
+     * @var LocaleHelper
+     */
+    private $localeHelper;
+
+    /**
+     * Constructor.
+     *
+     * @param $localeHelper
+     */
+    public function __construct(LocaleHelper $localeHelper)
+    {
+        $this->localeHelper = $localeHelper;
+    }
+
+    /**
+     * Transforms a date array into a calendar-aware localized date array.
+     *
+     * @param array $date Normalized date array.
+     *
+     * @return array Translated localized date array.
+     */
+    public function transform($date)
+    {
+        if (!is_array($date)) {
+            throw new TransformationFailedException('Expected an array.');
+        }
+
+        if ($this->localeHelper->getCurrentCalendar() !== 'gregorian') {
```

Comment:
This ``'gregorian'`` word appears many times in your code, maybe you should create some constants with calendars' names?